cpp/src/parquet/encoding-benchmark.cc

```diff
@@ -163,4 +172,132 @@ static void BM_DictDecodingInt64_literals(benchmark::State& state) {
 
 BENCHMARK(BM_DictDecodingInt64_literals)->Range(1024, 65536);
 
+std::shared_ptr<::arrow::Array> MakeRandomStringsWithRepeats(size_t num_unique,
+                                                             size_t num_values) {
+  const int64_t min_length = 2;
+  const int64_t max_length = 10;
+  std::vector<uint8_t> buffer(max_length);
+  std::vector<std::string> dictionary(num_unique);
+
+  uint32_t seed = 0;
+  std::default_random_engine gen(seed);
+  std::uniform_int_distribution<int64_t> length_dist(min_length, max_length);
+
+  std::generate(dictionary.begin(), dictionary.end(), [&] {
+    auto length = length_dist(gen);
+    ::arrow::random_ascii(length, seed++, buffer.data());
+    return std::string(buffer.begin(), buffer.begin() + length);
+  });
+
+  std::uniform_int_distribution<int64_t> indices_dist(0, num_unique - 1);
+  ::arrow::StringBuilder builder;
+
+  for (size_t i = 0; i < num_values; i++) {
+    const auto index = indices_dist(gen);
+    const auto value = dictionary[index];
+    ABORT_NOT_OK(builder.Append(value));
+  }
+
+  std::shared_ptr<::arrow::Array> result;
+  ABORT_NOT_OK(builder.Finish(&result));
+  return result;
+}
+
+class BM_PlainDecodingByteArray : public ::benchmark::Fixture {
+ public:
+  void SetUp(const ::benchmark::State& state) override {
+    num_values_ = static_cast<int>(state.range());
+
+    input_array_ = MakeRandomStringsWithRepeats(num_values_ / 8, num_values_);
+    const auto& binary_array = static_cast<const ::arrow::BinaryArray&>(*input_array_);
+    values_ = std::vector<ByteArray>();
+    values_.reserve(num_values_);
+    total_size_ = 0;
+
+    for (int64_t i = 0; i < binary_array.length(); i++) {
```

Comment:
According to my understanding, lines 212-229 are taking the random `arrow::Array` generated by `MakeRandomStringsWithRepeats` and "plain encoding" it as a Parquet `ByteArray`, with the intent that this `ByteArray` will be used as the test data for measuring the performance of decoding in the benchmark functions associated with this `benchmark::Fixture`.

Assuming my understanding is correct, consider factoring lines 212-229 into a helper function like `PlainEncodeByteArray(std::shared_ptr<::arrow::Array> array)`, or similar, to help make clear what the fixture `SetUp` process is doing at a high level. It would be nice to be able to glance at this `SetUp` code in order to understand the pre-conditions of the benchmark functions.

Reply:
You understanding is correct and your suggestion makes this clearer.